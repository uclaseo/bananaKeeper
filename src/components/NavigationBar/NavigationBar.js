import React, { PureComponent, Fragment } from 'react';
import { Link } from 'react-router-dom';
import {
  AppBar,
  Toolbar,
  IconButton,
  Drawer,
  List,
  ListItem,
  ListItemIcon,
  ListItemText,
  Divider,
} from '@material-ui/core';

import {
  Home,
  AddCircle,
  RemoveCircle,
  Assessment,
  Menu,
} from '@material-ui/icons';

export default class NavigationBar extends PureComponent {
  state = {
    isOpen: false,
  };

  toggleDrawer = () => {
    this.setState((state) => {
      return {
        isOpen: !state.isOpen,
      };
    });
  }

  renderMenu = () => {
    return (
      <div
        role="button"
        tabIndex={0}
        onClick={this.toggleDrawer}
        onKeyDown={this.toggleDrawer}
        style={{
          width: 250,
        }}
      >

        <Link to="/" style={{ textDecoration: 'none' }}>
          <List>
            <ListItem
              button
            >
              <ListItemIcon>
                <Home />
              </ListItemIcon>
              <ListItemText primary="Home" />
            </ListItem>
          </List>
        </Link>

        <Divider />

        <Link to="buy" style={{ textDecoration: 'none' }}>
          <List>
            <ListItem
              button
            >
              <ListItemIcon>
                <AddCircle />
              </ListItemIcon>
              <ListItemText primary="Buy" />
            </ListItem>
          </List>
        </Link>

        <Link to="sell" style={{ textDecoration: 'none' }}>
          <List>
            <ListItem
              button
            >
              <ListItemIcon>
                <RemoveCircle />
              </ListItemIcon>
              <ListItemText primary="Sell" />
            </ListItem>
          </List>
        </Link>

        <Link to="analytics" style={{ textDecoration: 'none' }}>
          <List>
            <ListItem
              button
            >
              <ListItemIcon>
                <Assessment />
              </ListItemIcon>
              <ListItemText primary="Analytics" />
            </ListItem>
          </List>
        </Link>

      </div>
    );
  }

  render() {
    const { isOpen } = this.state;
    return (
      <Fragment>

        <AppBar
          position="static"
          color="primary"
        >
          <Toolbar>
            <IconButton
              onClick={this.toggleDrawer}
              color="inherit"
              aria-label="Menu"
            >
              <Menu />
            </IconButton>
          </Toolbar>
        </AppBar>

        <Drawer
          open={isOpen}
          onClose={this.toggleDrawer}
        >
          {this.renderMenu()}
        </Drawer>

      </Fragment>
    );
  }
}
