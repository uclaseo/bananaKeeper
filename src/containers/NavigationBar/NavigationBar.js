import React, { Component } from 'react';
import { Link } from 'react-router-dom';

export default class NavigationBar extends Component {
  render() {
    return (
      <div>
        <Link to='buy'>buy</Link>
        <Link to='sell'>sell</Link>
        <Link to='analytics'>analytics</Link>
      </div>
    );
  }
};