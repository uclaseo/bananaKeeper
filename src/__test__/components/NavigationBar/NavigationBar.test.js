import React from 'react';
import { create } from 'react-test-renderer';
import NavigationBar from '../../../components/NavigationBar/NavigationBar';

describe('NavigationBar component', () => {
  test('it matches the snapshot', () => {
    const component = create(<NavigationBar />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
