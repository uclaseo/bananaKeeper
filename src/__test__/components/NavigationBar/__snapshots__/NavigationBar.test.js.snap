// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationBar component it matches the snapshot 1`] = `
<header
  className="MuiPaper-root-10 MuiPaper-elevation4-16 MuiAppBar-root-1 MuiAppBar-positionStatic-5 MuiAppBar-colorPrimary-8"
>
  <div
    className="MuiToolbar-root-37 MuiToolbar-regular-39 MuiToolbar-gutters-38"
  >
    <button
      aria-label="Menu"
      className="MuiButtonBase-root-47 MuiIconButton-root-41 MuiIconButton-colorInherit-42"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex="0"
      type="button"
    >
      <span
        className="MuiIconButton-label-46"
      >
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root-50"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
          <path
            d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
          />
        </svg>
      </span>
      <span
        className="MuiTouchRipple-root-73"
      />
    </button>
  </div>
</header>
`;
